src/lib/renderer.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from "react";
+import ReactDOM from "react-dom";
+import apps from "../apps";
+
+function renderApps(query: string) {
+  apps.every(async (app) => {
+    console.log("checking render for app", app);
+    // Question: Not all apps need to return data.
+    // TODO: Clearly define what is success and what is failure criteria for these apps somewhere.
+    const data = await app.queryToData(query);
```

Comment:
@OrkoHunter we pass `{query}`, as a key inside object, instead of flat `query` to queryToData, as it allows us to pass additional parameters to the queryToData in future without breaking the API

Reply:
Yeah, I was thinking about this. Was hoping the API could look like

```
queryToData(query, options)
```

Where options might contain optional things like necessary tokens, environment variables, and what not.

But it's fine to have an extendible object as the only parameter. Will revert!

Reply:
Done!